File: ruby/sum-of-multiples/sum_of_multiples.rb
class SumOfMultiples
  private

  attr_reader :factors

  public

  def initialize(*factors)
    @factors = factors
  end

  def to(limit)
    1 .upto(limit - 1)
      .select { |n| factors.any? { |f| (n % f).zero? } }
      .sum
  end
end

File: lua/sum-of-multiples/sum-of-multiples.lua
local sum_of_multiples = function(factors)

    local is_a_multiple = function (n)
        for _, f in pairs(factors) do
            if f ~= 0 and n % f == 0 then
                return true
            end
        end
        return false
    end

    local to = function (n)
        local sum = 0
        for i = 1, n-1 do
            if is_a_multiple(i) then
                sum = sum + i
            end
        end
        return sum
    end

    return { to = to }
end

return sum_of_multiples
